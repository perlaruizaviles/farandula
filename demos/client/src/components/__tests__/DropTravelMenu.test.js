import React from 'react';
import {shallow, mount} from 'enzyme';
import DropTravelMenu from '../DropTravelMenu';
import {Map, List} from 'immutable';

function setup(){
    const props = {
        config: Map({
            cabin: 'economy',
            travelers:Map({
                'adults': 1,
                'seniors': 0,
                'youth': 0,
                'child': 0,
                'seat-infant': 0,
                'lap-infant': 0
            })
        }),
        options: Map({cabin: List(['economy', 'premium-economy', 'business', 'first'])}),
        travelerTypeCountChange: () => {},
        cabinChange: () => {}
    }
    return shallow(<DropTravelMenu {...props} />);
}

describe('Rendering DropTravelMenu ', () => {
  it('Renders Dropdown', () => {
    const wrapper = setup();
    expect(wrapper.find('Dropdown').length).toBe(1);
  });
});