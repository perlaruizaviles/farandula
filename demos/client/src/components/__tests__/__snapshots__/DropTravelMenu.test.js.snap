exports[`Rendering DropTravelMenu Should create an snapshot for DropTravelMenu 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={false}
    closeOnChange={false}
    floating={true}
    icon="dropdown"
    noResultsMessage="No results found."
    openOnFocus={true}
    pointing={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    style={
        Object {
            "width": 200,
          }
    }
    text="1 Adult, Economy">
    <DropdownMenu
        onClick={[Function]}>
        <DropdownHeader
            content="Cabin" />
        <DropdownItem>
            <DropCabinMenu
                cabinChange={[Function]}
                config={
                    Object {
                        "cabin": "economy",
                        "travelers": Object {
                          "adults": 1,
                          "child": 0,
                          "lap-infant": 0,
                          "seat-infant": 0,
                          "seniors": 0,
                          "youth": 0,
                        },
                      }
                }
                options={
                    Object {
                        "cabin": Array [
                          "economy",
                          "premium-economy",
                          "business",
                          "first",
                        ],
                      }
                } />
        </DropdownItem>
        <DropdownDivider />
        <DropdownHeader
            content="Passengers" />
        <Segment
            basic={true}>
            <TravelerMenu
                config={
                    Object {
                        "cabin": "economy",
                        "travelers": Object {
                          "adults": 1,
                          "child": 0,
                          "lap-infant": 0,
                          "seat-infant": 0,
                          "seniors": 0,
                          "youth": 0,
                        },
                      }
                }
                options={
                    Object {
                        "cabin": Array [
                          "economy",
                          "premium-economy",
                          "business",
                          "first",
                        ],
                      }
                }
                travelerTypeCountChange={[Function]} />
        </Segment>
    </DropdownMenu>
</Dropdown>,
  "nodes": Array [
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={false}
      closeOnChange={false}
      floating={true}
      icon="dropdown"
      noResultsMessage="No results found."
      openOnFocus={true}
      pointing={true}
      renderLabel={[Function]}
      selectOnBlur={true}
      style={
            Object {
                  "width": 200,
                }
      }
      text="1 Adult, Economy">
      <DropdownMenu
            onClick={[Function]}>
            <DropdownHeader
                  content="Cabin" />
            <DropdownItem>
                  <DropCabinMenu
                        cabinChange={[Function]}
                        config={
                              Object {
                                    "cabin": "economy",
                                    "travelers": Object {
                                      "adults": 1,
                                      "child": 0,
                                      "lap-infant": 0,
                                      "seat-infant": 0,
                                      "seniors": 0,
                                      "youth": 0,
                                    },
                                  }
                        }
                        options={
                              Object {
                                    "cabin": Array [
                                      "economy",
                                      "premium-economy",
                                      "business",
                                      "first",
                                    ],
                                  }
                        } />
            </DropdownItem>
            <DropdownDivider />
            <DropdownHeader
                  content="Passengers" />
            <Segment
                  basic={true}>
                  <TravelerMenu
                        config={
                              Object {
                                    "cabin": "economy",
                                    "travelers": Object {
                                      "adults": 1,
                                      "child": 0,
                                      "lap-infant": 0,
                                      "seat-infant": 0,
                                      "seniors": 0,
                                      "youth": 0,
                                    },
                                  }
                        }
                        options={
                              Object {
                                    "cabin": Array [
                                      "economy",
                                      "premium-economy",
                                      "business",
                                      "first",
                                    ],
                                  }
                        }
                        travelerTypeCountChange={[Function]} />
            </Segment>
      </DropdownMenu>
</Dropdown>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <DropTravelMenu
        cabinChange={[Function]}
        config={
                Object {
                        "cabin": "economy",
                        "travelers": Object {
                          "adults": 1,
                          "child": 0,
                          "lap-infant": 0,
                          "seat-infant": 0,
                          "seniors": 0,
                          "youth": 0,
                        },
                      }
        }
        options={
                Object {
                        "cabin": Array [
                          "economy",
                          "premium-economy",
                          "business",
                          "first",
                        ],
                      }
        }
        travelerTypeCountChange={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "cabinChange": [Function],
          "config": Object {
            "cabin": "economy",
            "travelers": Object {
              "adults": 1,
              "child": 0,
              "lap-infant": 0,
              "seat-infant": 0,
              "seniors": 0,
              "youth": 0,
            },
          },
          "options": Object {
            "cabin": Array [
              "economy",
              "premium-economy",
              "business",
              "first",
            ],
          },
          "travelerTypeCountChange": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={false}
          closeOnChange={false}
          floating={true}
          icon="dropdown"
          noResultsMessage="No results found."
          openOnFocus={true}
          pointing={true}
          renderLabel={[Function]}
          selectOnBlur={true}
          style={
                    Object {
                              "width": 200,
                            }
          }
          text="1 Adult, Economy">
          <DropdownMenu
                    onClick={[Function]}>
                    <DropdownHeader
                              content="Cabin" />
                    <DropdownItem>
                              <DropCabinMenu
                                        cabinChange={[Function]}
                                        config={
                                                  Object {
                                                            "cabin": "economy",
                                                            "travelers": Object {
                                                              "adults": 1,
                                                              "child": 0,
                                                              "lap-infant": 0,
                                                              "seat-infant": 0,
                                                              "seniors": 0,
                                                              "youth": 0,
                                                            },
                                                          }
                                        }
                                        options={
                                                  Object {
                                                            "cabin": Array [
                                                              "economy",
                                                              "premium-economy",
                                                              "business",
                                                              "first",
                                                            ],
                                                          }
                                        } />
                    </DropdownItem>
                    <DropdownDivider />
                    <DropdownHeader
                              content="Passengers" />
                    <Segment
                              basic={true}>
                              <TravelerMenu
                                        config={
                                                  Object {
                                                            "cabin": "economy",
                                                            "travelers": Object {
                                                              "adults": 1,
                                                              "child": 0,
                                                              "lap-infant": 0,
                                                              "seat-infant": 0,
                                                              "seniors": 0,
                                                              "youth": 0,
                                                            },
                                                          }
                                        }
                                        options={
                                                  Object {
                                                            "cabin": Array [
                                                              "economy",
                                                              "premium-economy",
                                                              "business",
                                                              "first",
                                                            ],
                                                          }
                                        }
                                        travelerTypeCountChange={[Function]} />
                    </Segment>
          </DropdownMenu>
</Dropdown>,
        "_debugID": 2,
        "_renderedOutput": <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={false}
          closeOnChange={false}
          floating={true}
          icon="dropdown"
          noResultsMessage="No results found."
          openOnFocus={true}
          pointing={true}
          renderLabel={[Function]}
          selectOnBlur={true}
          style={
                    Object {
                              "width": 200,
                            }
          }
          text="1 Adult, Economy">
          <DropdownMenu
                    onClick={[Function]}>
                    <DropdownHeader
                              content="Cabin" />
                    <DropdownItem>
                              <DropCabinMenu
                                        cabinChange={[Function]}
                                        config={
                                                  Object {
                                                            "cabin": "economy",
                                                            "travelers": Object {
                                                              "adults": 1,
                                                              "child": 0,
                                                              "lap-infant": 0,
                                                              "seat-infant": 0,
                                                              "seniors": 0,
                                                              "youth": 0,
                                                            },
                                                          }
                                        }
                                        options={
                                                  Object {
                                                            "cabin": Array [
                                                              "economy",
                                                              "premium-economy",
                                                              "business",
                                                              "first",
                                                            ],
                                                          }
                                        } />
                    </DropdownItem>
                    <DropdownDivider />
                    <DropdownHeader
                              content="Passengers" />
                    <Segment
                              basic={true}>
                              <TravelerMenu
                                        config={
                                                  Object {
                                                            "cabin": "economy",
                                                            "travelers": Object {
                                                              "adults": 1,
                                                              "child": 0,
                                                              "lap-infant": 0,
                                                              "seat-infant": 0,
                                                              "seniors": 0,
                                                              "youth": 0,
                                                            },
                                                          }
                                        }
                                        options={
                                                  Object {
                                                            "cabin": Array [
                                                              "economy",
                                                              "premium-economy",
                                                              "business",
                                                              "first",
                                                            ],
                                                          }
                                        }
                                        travelerTypeCountChange={[Function]} />
                    </Segment>
          </DropdownMenu>
</Dropdown>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DropTravelMenu
    cabinChange={[Function]}
    config={
        Object {
            "cabin": "economy",
            "travelers": Object {
              "adults": 1,
              "child": 0,
              "lap-infant": 0,
              "seat-infant": 0,
              "seniors": 0,
              "youth": 0,
            },
          }
    }
    options={
        Object {
            "cabin": Array [
              "economy",
              "premium-economy",
              "business",
              "first",
            ],
          }
    }
    travelerTypeCountChange={[Function]} />,
}
`;
